### PR TITLE
add double gaussian magnification_finite computation and plot_quasar_images 

### PR DESCRIPTION
**Pull request by dangilman**
I added the option to use two Gaussian source light profiles in the magnification_finite_adaptive computation, and the routine plot_quasar_images which produces plots of individual quasar images like the ones I often show

### CODE REVIEW

**Comment by dangilman** [comment]
Two things: I removed the @jit decorators in my branch, that might be included in the pull request, maybe I should get rid of that. Also the test function for plot_quasar_images produces a plot, not sure if that will make the build fail

File: lenstronomy/LensModel/lens_model_extensions.py [hunk 1]
```diff
@@ -25,25 +25,132 @@ def hessian()
         """
         self._lensModel = lensModel
 
+    def plot_quasar_images(self, x_image, y_image, source_x, source_y, kwargs_lens,
```
[the remark this answers is not in the document]
Comment (sibirrer) [reply lenstronomy/LensModel/lens_model_extensions.py:28]: this plotting function has a lot of calculations/ lines of code not specific to an execution of a plotting routine. Perhaps splitting these tasks might help with the code design. Just a though

**Review by sibirrer — CHANGES_REQUESTED** [review]
my main comment is that plotting routines and analysis/calculations should be well separated (as good as possible), such that the there is not much duplication of code and routines and the plotting is clearly separated from a user API. Thanks!

File: lenstronomy/Util/magnification_finite_util.py [hunk 2]
```diff
@@ -24,4 +29,69 @@ def auto_raytracing_grid_resolution(source_fwhm_parcsec, grid_resolution_scale=0
     """
 
     grid_resolution = grid_resolution_scale * (source_fwhm_parcsec / ref) ** power
-    return grid_resolution
\ No newline at end of file
+    return grid_resolution
+
+def setup(cosmo, lens_model, grid_radius_arcsec, grid_resolution, source_fwhm_parsec, source_light_model, z_source,
```
Comment (sibirrer) [inline lenstronomy/Util/magnification_finite_util.py:34]: this routine name is a bit too unspecific. Could you add some naming that is better distinguishable from other routines and a bit more specific?

**Review by sibirrer — CHANGES_REQUESTED** [review]
I have one minor request, otherwise looks great! Just waiting for the tests to pass too. Thanks

**Review by sibirrer — APPROVED** [review]

**Review by sibirrer — CHANGES_REQUESTED** [review]
Very minor (pedantic) request to make for an optimal test coverage: can you test the TORUS and raise statements too? I would like to approach 100% test coverage and all new lines of code being added to be tested. Thank you very much!

**Comment by dangilman** [comment]
I just removed it since there is no way that anyone could have been using magnification_finite with shape='TORUS'. The required keywords could not have been specified for the source light model, so it would crashed.

**Review by sibirrer — APPROVED** [review]

**Pull request merged**